src/storage/mod.rs

```diff
@@ -541,7 +545,7 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         start_ts: TimeStamp,
         key_only: bool,
         reverse_scan: bool,
-    ) -> impl Future<Item = Vec<Result<KvPair>>, Error = Error> {
+    ) -> impl Future<Output = Result<Vec<Result<KvPair>>>> {
```

Comment:
Could we write it as following:
```
pub async fn scan(...) -> Result<Vec<Result<KvPair>>> {
```

Reply:
How to infer the lifetime for `self` in async context.